src/KintoID.sol

```diff
@@ -484,21 +491,22 @@ contract KintoID is
 
     /**
      * @dev Hook that is called before any token transfer. Allow only mints and burns, no transfers.
-     * @param from source address
      * @param to target address
-     * @param batchSize The first id
+     * @param firstTokenId The first id
      */
-    function _beforeTokenTransfer(address from, address to, uint256 firstTokenId, uint256 batchSize)
+    function _update(address to, uint256 firstTokenId, address auth)
         internal
         virtual
         override(ERC721Upgradeable, ERC721EnumerableUpgradeable)
+        returns (address)
     {
+        address from = _ownerOf(firstTokenId);
         require(
             (from != address(0) && recoveryTargets[from] == to && isSanctionsSafe(from))
                 || (from == address(0) && to != address(0)) || (from != address(0) && to == address(0)),
             "Only recovery, mint or burn transfers are allowed"
         );
-        super._beforeTokenTransfer(from, to, firstTokenId, batchSize);
+        return super._update(to, firstTokenId, auth);
```

Comment:
do we need to call increaseBalance as per your comment above?

Reply:
So, this is what the changelog says:
```
- `ERC721`: The internal function `_beforeTokenTransfer` no longer updates balances, which it previously did when `batchSize` was greater than 1. This change has no consequence unless a custom ERC721 extension is explicitly invoking `_beforeTokenTransfer`. Balance updates in extensions must now be done explicitly using `__unsafe_increaseBalance`, with a name that indicates that there is an invariant that has to be manually verified.

```

Note the following: "This change has no consequence unless a **custom ERC721 extension is explicitly invoking** `_beforeTokenTransfer`". As far as I can see, even though when we override `_beforeTokenTransfer` we are kind of explicitly invoking `super._beforeTokenTransfer`, we do that with the same params we received. `batchSize` param is called internally on ERC721Upgradeable.sol with value '1'.

Does this make sense @rrecuero ?